src/checker.rs

```diff
@@ -49,7 +50,50 @@ mod tests {
         let profile = "release";
 
         let expected = format!("{}/target/{}/deps", pwd, profile);
-        assert_eq!(expected, get_base_path().unwrap());
+        let exe = env::current_exe();
+
+        match exe {
+            Ok(path) => assert_eq!(expected, get_base_path(path).unwrap()),
+            Err(_) => assert_eq!(true, false),
+        };
+    }
+
+    #[test]
+    fn test_download_and_unpack() -> Result<()> {
+        let base_url = generate_base_url("2.0.3");
+        let exe = env::current_exe();
```

Comment:
If you `unwrap()` here, you can drop the match on `exe` and don't need to make `base_path` mutable, since you can directly assign the result of `get_base_path`.